docs/adr/0037-use-nx-for-monorepo.md

```diff
@@ -0,0 +1,48 @@
+# Use nx for monorepo tooling
+
+- Status: proposed
+- Deciders: Ben Bangert, Wil Clouser, Dan Schomburg
+- Date: 2023-03-13
+- Obsoletes: 0019-use-workspace-dependencies
+
+## Context and Problem Statement
+
+Our monorepo using a yarn workspace model has a few limitations we've experienced such as slow builds, using bash scripts for running tasks, having to make each new library or component as its own package with its own dependencies, and an inflexible structure for refactoring our shared libraries.
+
+Monorepo tooling has progressed since the prior ADR on using workspace dependencies with yarn and we'd like to take advantage of these improvements.
+
+## Decision Drivers
+
+- Faster builds with dependency structure analysis and build caching.
+- Easy to migrate into without large changes to our existing setup.
+- Tooling built for monorepo needs.
+- Task running, caching, and dependency handling
+- Easy to create/share/move library/app code for re-use and encapsulation.
+
+## Considered Options
+
+- A. nx
+- B. Turborepo
+
+## Decision Outcome
+
+Chosen option: A. nx. Both options provide powerful build capabilities with similar caching options. nx is a little ahead on additional monorepo tooling that would benefit some of our upcoming work. nx supports two models of monorepo, a package-based approach (what FxA current is built with), and an integrated approach that utilizes TypeScript paths for easily splitting out and re-organizing code. This allows it to work almost as-is with our existing monorepo. We can then integrate/refactor existing code into lighter-weight integrated-style components for use in our applications.
+
+## Pros and Cons of the Options
+
+### A. nx
+
+- Good, because it has caching of tasks for fast builds.
+- Good, because its package-based approach works as-is on our monorepo with yarn workspaces.
+- Good, because its integrated libraries can be used in our existing packages with minimal changes.
+- Good, because its well supported with good documentation and commercial support available.
+- Good, because it has commands to move/delete/add libraries and update references automatically.
+- Good, because it has generators to streamline new library/application creation for consistency.
+- Bad, because its a new tool to learn for builds.
+
+### B. Turborepo
+
+- Good, because it has caching of tasks for fast builds.
+- Good, because its well supported with good documentation and commercial support available.
```

Comment:
This is interesting point. It appears turborepo is backed by vercel, which is likely a good thing. Who is Nx backed by? Looking at github stats alone nx seems better. It is used by almost 4x as many projects, has twice the total contributors, and appears more active.

Reply:
Nx is backed by the Nrwl company, which semes to have a decent team size, https://nx.app/company/